src/main/java/io/seqera/migtool/App.java

```diff
@@ -51,7 +52,8 @@ public Integer call() {
                 .withDialect(dialect!=null ? dialect : dialectFromUrl(url))
                 .withDriver(driver!=null ? driver : driverFromUrl(url))
                 .withLocations(location)
-                .withPattern(pattern);
+                .withPattern(pattern)
+                .withBuilder(SqlTemplate.from(dialect!=null ? dialect : dialectFromUrl(url)));
```

Comment:
I would prefer to let the details to be managed internally, after setting up the properties, rather than having conditionals to properly instantiate it...

Reply:
Do you mean not exposing `withBuilder` and keeping it as a side-effect of `withDialect`?

Reply:
Using your version, a user of MigTool needs to know how to setup the dialect, driver and the builder and check for nulls manually before setting them up.
Using my version, the proper instantiation is abstracted using the `Builder`, the user does not need to know how the internals work or are set. And the `side-effect` as you call it is basically what the builder pattern is used for. 

Now let's see the use case from the `SeqeraComputeManagerService`:

I need to instantiate Migtool from there, since we will use a custom `Datasource` and we are not using a CLI to instantiate it.
If I use your version, I would need to repeat all the conditionals again from the outside which assumes that I know how the internals of the MigTool work. If we let that work be done internally in MigTool, I don't need (nor should I need) to worry about that. 

<img width="1038" alt="image" src="https://github.com/user-attachments/assets/9cee2876-3724-4037-828a-dbb34d3d51db">


Finally, how is it better to have a static Helper class with methods that are respective to the Dialect and Driver only? And why return Strings that represent both the Dialect and Driver while we can type these and be more explicit and safe at the same time? While we can mixup Strings as a parameter of a method, we cannot mix the typed variable. And it is better to keep the behaviour as close as possible to the class to whom it belongs, we do not need a static Helper for that. It just makes the code harder to understand and expand.

https://github.com/seqeralabs/migtool/blob/c131b381d41ed9b12a02152372ce88c9718334c2/src/main/java/io/seqera/migtool/Helper.java#L204-L220

I think these changes are for the better and in fact they are more aligned with good programming practices, so I don't see any reason for this back and forth. I would like to kindly ask you to not block this PR because of this. If you compare the current code with the code from this PR, you will see that it is more intuitive and easier to use now. :)

Reply:
Thanks for a good tip to not expose the configuration for the builder/template. it can be inferred from the dialect setting. 

Regarding the other comment, I believe are out of comment. The goal of the issue is to add the support for Postgre, not to alter the Migtool public API

Reply:
> Thanks for a good tip to not expose the configuration for the builder/template. it can be inferred from the dialect setting.

👍🏻 



> Regarding the other comment, I believe are out of comment. The goal of the issue is to add the support for Postgre, not to alter the Migtool public API

Even though it was not my objective to change the API, it is a very small change and people can still use the old API with version 1.5.1, only the service that I am developing needs Postgres. But again, the way it is done is misleading since it is not actually a builder pattern. Why can't we improve this, since we are here? I implemented the Builder because to infer the `Dialect` and `Driver` from the `Url`, we need the Url set first (as it is now), otherwise we get a NullPtrException, so we are not helping the users of MigTool here. Having this type of code that breaks easily is not good imo.

Reply:
Think we discussed, a PR should be scope in the the goal it declares and should not introduce unrelated changes for the sake of change tracking and effectiveness of the review.